### DISCUSSION TITLE
Custom commit message

### DISCUSSION THREAD

**Discussion by craigfurman** [GitJournal/GitJournal#686] [Feature Requests]
I'd like to be able to configure a custom commit message in gitjournal settings, to override the "Edited Note" default. My exact use case: I write most of my notes on desktop with Obsidian, and use GitJournal to read and write the same set of notes on mobile. I'd customise the commit message to "Edited via GitJournal".

This is a pretty low priority for me tbh, but figured it was worth raising anyway! I think this is a distinct issue from https://github.com/GitJournal/GitJournal/issues/201, because this issue proposes configuring a static message rather than dynamically forming a message based on the content of the commit.

WDYT?

**Comment by vHanda** [comment]
Hey Craig. Cool beard!

A hacky way of doing this could be to change the "Git Author" in GitJournal to say "GitJournal" so it's easier to notice in the git log.

I'm open to letting users change this, but it would require -
1. A UI to be built to expose each of these - https://github.com/GitJournal/GitJournal/blob/master/lib/core/commit_message_builder.dart
2. Some kind of templating language to allow the current way.

If you can figure out (2), I can probably whip up a simple UI in a hour or so. It would be nice if the templating language is compatible with Obsidian's template syntax and if it could also be used to give users more control over how the filename should be stored.

**Comment by craigfurman** [comment]
> Cool beard!

Thanks dude!

https://github.com/GitJournal/GitJournal/blob/master/lib/core/commit_message_builder.dart - interesting, but I'm confused: in the android app, all of my commit messages seem to be a static "Edited Note", and I see no way to change this in config. In what context is the logic in that file run?

> Some kind of templating language to allow the current way

My proposal is to allow global override via app config of the commit message to a static string, no templating required.
